treenode_test.go

```diff
@@ -194,3 +210,55 @@ func (s *spawnProto) HandlerError2(msg spawnMsg) error {
 func (s *spawnProto) HandlerError3(msg spawnMsg) (int, error) {
 	return 0, nil
 }
+
+// Simple protocol to have messages transmit between nodes
+const dummyName = "TestDummy"
+
+type DummyProtoMsg struct{}
+
+type dummyProto struct {
+	*TreeNodeInstance
+	done      chan bool
+	dummyChan chan struct {
+		*TreeNode
+		DummyProtoMsg
+	}
+}
+
+var dummyDone = make(chan bool)
```

Comment:
instead of using a global channel, you can make a channel in the `dummyProto` struct, and then write to it when Dispatch is done on root.  Actually, you already have it - `done chan bool`.

Then just write to the channel in dispatch like 
```

func (cp *dummyProto) Dispatch() error {
	select {
	defer cp.Done()
	case <-cp.dummyChan:
		if !cp.IsRoot() { cp.SendToParent(&DummyProtoMsg{}) }
	}
	cp.done  <- true // probably needs to be buffered so that the children can return
	return nil
} 
```
I think that should work...

Reply:
Yes, and that's much cleaner, thanks!